src/main.py

```diff
@@ -11,17 +13,51 @@
 # while developing.
 BROWSABLE = os.environ.get("LOCAL_BROWSE_API")
 
-app = FastAPI()
+# Create a FastAPI instance with metadata
+app = FastAPI(
+    title="IDPD-API-POC",
+    description="A proof of concept API layer to manage content negotiation across IDPD resources and provide an abstraction to 3rd party data services. This API provides access to various metadata, including datasets, editions, publishers, and topics.",
+    version="0.0.1",
+)
 
 
-@app.get("/datasets", response_model=Optional[schemas.Datasets])
-def datasets(
+@app.get(
```

Comment:
so the statement `Optional` means "the thing in brackets **or** a None/Null response". So if our response model is `Optional[schemas.Datasets]` then any given documented response should be specifically returning one of those two options.

by way of an example, written like this...

```
@app.get(
    "/datasets", 
    response_model=Optional[schemas.Datasets], 
    responses={
        status.HTTP_200_OK: {
            "description": "Successful response. Returns a list of datasets.",
            "model": schemas.Datasets,
        },
        status.HTTP_404_NOT_FOUND: {
            "description": "Not Found. No datasets found.",
            "model": None,
        },
        status.HTTP_406_NOT_ACCEPTABLE: {
            "description": "Not Acceptable. The requested format is not supported.",
            "model": None,
        },
    }
)
```

the responses match the response model info, at the moment it's a bit dy-syncced.

could you apply this same logic across the other endpoints as well please.